tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -0,0 +1,255 @@
+sub set_mutex_lock {
+    my ($self, $lock_signal) = @_;
+
+    mutex_lock($lock_signal);
+    mutex_unlock($lock_signal);
+}
+
+sub rke2_server_setup {
+    my ($self, $agent_ip) = @_;
+
+    systemctl('stop apparmor.service');
+    # systemctl('stop SuSEfirewall2.service');
+    assert_script_run('ssh-keygen -b 2048 -t rsa -q -N "" -f ~/.ssh/id_rsa') unless (-e "~/.ssh/id_rsa");
+    exec_and_insert_password("ssh-copy-id -o StrictHostKeyChecking=no -i /root/.ssh/id_rsa.pub root\@$agent_ip");
+
+    # Ensure SUSE certificates are installed on the node
+    ensure_ca_certificates_suse_installed();
+
+    # Install downstream kubevirt packages
+    zypper_call('in kubernetes1.18-client') if (script_run('rpmquery kubernetes1.18-client'));
+
+    # RKE2 deployment on server node
+    my $rke_ver = get_var('INSTALL_RKE2_VERSION');
+    if ($rke_ver) {
+        assert_script_run("curl -sfL https://get.rke2.io | INSTALL_RKE2_VERSION=$rke_ver sh -");
+    } else {
+        assert_script_run('curl -sfL https://get.rke2.io | sh -');
+    }
+    systemctl('enable rke2-server.service');
+    systemctl('start rke2-server.service');
+
+    # Start rke2-server service ready
+    mutex_create('RKE2_SERVER_START_READY');
+
+    # Wait for rke2-agent service to be ready
+    $self->set_mutex_lock('RKE2_AGENT_START_READY');
+
+    assert_script_run('cp /etc/rancher/rke2/rke2.yaml ~/.kube/config');
+    assert_script_run('kubectl get nodes');
+
+    # Create registries ready
+    script_run("cat > /etc/rancher/rke2/registries.yaml <<__END
+mirrors:
+  registry.suse.de:
+    endpoint:
+      - http://registry.suse.de
+configs:
+  registry.suse.de:
+    tls:
+      insecure_skip_verify: true
+__END");
+
+    assert_script_run("scp /etc/rancher/rke2/registries.yaml root\@$agent_ip:/etc/rancher/rke2/registries.yaml");	
+    mutex_create('CREATE_REGISTRIES_READY');
+
+    # Restart RKE2 service and check the service is active well after restart
+    systemctl('restart rke2-server.service');
+    check_service_status();
+}
+
+sub check_service_status {
+    # Check RKE2 services & kubevirt test environment
+    systemctl('is-active rke2-server.service');
+}
+
+sub deploy_kubevirt_manifests {
+    my $os_version = script_output("cat /etc/os-release | grep VERSION= | awk -F\" '{print \$2}'");
+    zypper_call("ar http://download.suse.de/ibs/Devel:/Virt:/SLE-" . $os_version . "/SUSE_SLE-" . $os_version . "_Update_standard/ devel");
+    zypper_call('ref devel');
+    zypper_call('in -f -r devel kubevirt-manifests kubevirt-virtctl kubevirt-tests containerized-data-importer-manifests') if (script_run('rpmquery kubevirt-manifests kubevirt-virtctl kubevirt-tests containerized-data-importer-manifests'));
+    assert_script_run("rpm -qa |grep -E \"virt-test|kubevirt|containerized\"");
+
+    assert_script_run("kubectl apply -f /usr/share/cdi/manifests/release/cdi-operator.yaml");
+    assert_script_run("kubectl apply -f /usr/share/cdi/manifests/release/cdi-cr.yaml");
```

Comment:
kubectl -n kubevirt wait kv kubevirt --for condition=Available --timeout=${INSTALL_TIMEOUT}
fi

# CDI currently does not support downgrades. So delete it and deploy from scratch.
if kubectl -n cdi get cdis cdi; then
    kubectl delete -f ${CDI_CR}
    kubectl delete -f ${CDI_OPERATOR}
fi

You need to add below to ensure cdi deployment is really up and ready.
`kubectl -n cdi wait cdis cdi --for condition=Available --timeout=${INSTALL_TIMEOUT}`

Reply:
Added this script. Is the parameter ${INSTALL_TIMEOUT} from the runtime job or it's defined casually by our own?

Reply:
You can define it :)

Reply:
OK, the timeout is set with 15m.